source/gpus_in_openstack.rst

```diff
@@ -0,0 +1,313 @@
+To enable it add it to the list of filters to Kolla-Ansible configuration file:
+``etc/kayobe/kolla/config/nova.conf``, for instance:
+
+.. code-block:: yaml
+
+   [filter_scheduler]
+   available_filters = nova.scheduler.filters.all_filters
+   enabled_filters = AvailabilityZoneFilter, ComputeFilter, ComputeCapabilitiesFilter, ImagePropertiesFilter, ServerGroupAntiAffinityFilter, ServerGroupAffinityFilter, PciPassthroughFilter
+
+Configure nova-compute
+~~~~~~~~~~~~~~~~~~~~~~
+
+Configuration can be applied in flexible ways using Kolla-Ansible's
+methods for `inventory-driven customisation of configuration
+<https://docs.openstack.org/kayobe/latest/configuration/reference/kolla-ansible.html#service-configuration>`_.
+The following configuration could be added to
+``etc/kayobe/kolla/config/nova/nova-compute.conf`` to enable PCI
+passthrough of GPU devices for hosts in a group named ``compute_gpu``.
+Again, the 4-digit PCI Vendor ID and Device ID extracted from ``lspci
+-nn`` can be used here to specify the GPU device(s).
+
+.. code-block:: jinja
+
+   [pci]
+   {% raw %}
+   {% if inventory_hostname in groups['compute_gpu'] %}
+   # We could support multiple models of GPU.
+   # This can be done more selectively using different inventory groups.
+   # GPU models defined here:
+   # NVidia Tesla V100 16GB
+   # NVidia Tesla V100 32GB
+   # NVidia Tesla P100 16GB
+   passthrough_whitelist = [{ "vendor_id":"10de", "product_id":"1db4" },
+                            { "vendor_id":"10de", "product_id":"1db5" },
+                            { "vendor_id":"10de", "product_id":"15f8" }]
+   alias = { "vendor_id":"10de", "product_id":"1db4", "device_type":"type-PCI", "name":"gpu-v100-16" }
+   alias = { "vendor_id":"10de", "product_id":"1db5", "device_type":"type-PCI", "name":"gpu-v100-32" }
+   alias = { "vendor_id":"10de", "product_id":"15f8", "device_type":"type-PCI", "name":"gpu-p100" }
+   {% endif %}
+   {% endraw %}
+
+Configure nova-api
+~~~~~~~~~~~~~~~~~~
+
+pci.alias also needs to be configured on the controller. 
+This configuration should match the configuration found on the compute nodes.
+Add it to Kolla-Ansible configuration file:
+``etc/kayobe/kolla/config/nova-api/nova.conf``, for instance:
+
+.. code-block:: yaml
+
+   [pci]
+   alias = { "vendor_id":"10de", "product_id":"1db4", "device_type":"type-PCI", "name":"gpu-v100-16" }
+   alias = { "vendor_id":"10de", "product_id":"1db5", "device_type":"type-PCI", "name":"gpu-v100-32" }
+   alias = { "vendor_id":"10de", "product_id":"15f8", "device_type":"type-PCI", "name":"gpu-p100" }
+
+Reconfigure nova service
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: text
+
+   kayobe overcloud service reconfigure -kt nova --kolla-skip-tags common --skip-precheck
+
+Configure a flavor
+~~~~~~~~~~~~~~~~~~
+For example, to request two of the GPUs with alias gpu-p100
+
+.. code-block:: text
+
+   openstack flavor set m1.medium --property "pci_passthrough:alias"="gpu-p100:2"
```

Comment:
Can we make a reference here to the project-config repo, eg https://github.com/stackhpc/openstack-admin-guide/blob/master/source/vars.rst#L29